src/Route.php

```diff
@@ -170,10 +174,31 @@ public function defaults(array $defaults): self
 
     /**
      * @param callable|MiddlewareInterface $middleware
-     * @return MiddlewareInterface
+     * @return MiddlewareInterface|string|array
      */
-    private function prepareMiddleware($middleware): MiddlewareInterface
+    private function prepareMiddleware($middleware)
     {
+        if (
+            is_string($middleware) && class_exists($middleware)
+            && in_array(MiddlewareInterface::class, class_implements($middleware))
+        ) {
+            if ($this->container === null) {
+                throw new InvalidArgumentException('Route container must not be null for lazy loaded middleware.');
+            }
+            return $middleware;
+        }
+
+        if (
+            is_array($middleware) && count($middleware) === 2
+            && isset($middleware[0]) && isset($middleware[1])
```

Comment:
This check not needed

Reply:
What exactly?

Reply:
Because the next step is checked for a string
```php
$middleware = [null, 'index']; // false
$middleware = [false, 'index']; // false
$middleware = [true, 'index']; // false
```

Reply:
What do you mean exactly?
1. What condition will we remove.
2. What is the next step.

Reply:
1. Condition `isset` is not needed
2. I mean condition `is_string`

Reply:
No, you're wrong. 
Use case: $middleware = [null, 'index']; 
is_string($middleware[0])  //an exception will be thrown

Reply:
Oh, you're right. I'm so tired :(